spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/resource/ResourceController.java

```diff
@@ -71,87 +70,119 @@ public ResourceController(ResourceRepository resourceRepository,
 
 	@RequestMapping("/{name}/{profile}/{label}/**")
 	public String retrieve(@PathVariable String name, @PathVariable String profile,
-			@PathVariable String label, HttpServletRequest request,
+        	@PathVariable String label, ServletWebRequest request,
```

Comment:
I wonder about changing the signature on all of these public methods.

AFAICS you are using it for
```
request.checkNotModified(resource.lastModified());
```

Can you do 
```
new ServletWebRequest(HttpServletRequest);
```

in `checkNotModified()`?

It would simplify things and keep signatures.

Reply:
What is the advantage of that? Although the methods are public the only caller of them will be Spring MVC and the unit tests and Spring handles this automatically. As for simpler, I would argue that the current code is simpler because it doesn't have to have the new ServletWebRequest in every endpoint method. In addition, the only place the HttpServletRequest is used is in getFilePath(). So the real argument for doing what you are suggesting is to keep it backward compatible even though that doesn't really matter.

Reply:
We are very particular about any public api we publish. This being one of them.

Reply:
If Imake the requested change it will still cause the method signatures to be modified. ServletWebRequest must be constructed with the HttpServletRequest and HttpServletResponse. Currently the HttpServletResponse is not being passed to the methods. Without the HttpServletResponse ServletWebRequest can't set the HTTP status code that needs to be returned.

Reply:
There is a request only constructor

Reply:
Yup, and it is exactly equivalent to calling new ServletWebRequest(request, null) which won't work.

Reply:
I didn't dig into `checkNotModified()` to see that it required the response.

Reply:
All right, my team convinced me otherwise. You can remove the backwards compatible methods. If you don't have time, we can do it on merge.

Reply:
OK. I removed them.